### ISSUE TITLE
apt_repository: add 'arch' option

### ISSUE THREAD

**Issue by mapreri** [ansible/ansible#18975]
_From @mapreri on December 24, 2014 15:19_

##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME
apt_repository module
##### ANSIBLE VERSION
never worked…

##### SUMMARY
I'm my systems I have a lot of repositories with an arch-qualifier.
e.g.:

```
deb [arch=ppc64el] http://ftp.debian.org/debian/ sid main
deb [arch=amd64,i386] http://dl.google.com/linux/talkplugin/deb/ stable main
```

apt_repository does not understand them and tries to re-add that repos, actually duplicating them.

Moreover I'd like to be able to specify these option in ansible. Currently available options are (from sources.list(5):

```
       ·   arch=arch1,arch2,...  can be used to specify for which architectures information should be downloaded. If this option is not set all architectures
           defined by the APT::Architectures option will be downloaded.

       ·   arch+=arch1,arch2,...  and arch-=arch1,arch2,...  which can be used to add/remove architectures from the set which will be downloaded.

       ·   trusted=yes can be set to indicate that packages from this source are always authenticated even if the Release file is not signed or the signature can't
           be checked. This disables parts of apt-secure(8) and should therefore only be used in a local and trusted context.  trusted=no is the opposite which
           handles even correctly authenticated sources as not authenticated.
```

these options are surrounded by square brackets, and can or cannot have leading and trailing spaces between the options and the brackets.

_Copied from original issue: ansible/ansible-modules-core#563_

**Comment by mapreri** [comment]
_From @ansibot on August 1, 2016 3:45_



@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by mapreri** [comment]
_From @ansibot on September 13, 2016 2:0_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mapreri** [comment]
_From @ansibot on October 6, 2016 16:44_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mapreri** [comment]
_From @ansibot on October 22, 2016 9:36_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mapreri** [comment]
_From @ansibot on December 7, 2016 18:14_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by mapreri** [comment]
BTW, starting from Debian stretch there will be a new way to specify repositories, namely deb822-formatted sources entries.  See [the relevant manpage](https://manpages.debian.org/cgi-bin/man.cgi?query=sources.list&apropos=0&sektion=5&manpath=Debian+unstable+sid&format=html&locale=en) for info.  It would be pretty cool to support that too.

**Comment by ansibot** [comment]
@mapreri Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mapreri** [comment]
srly, guys....
(done, hopefully)

**Comment by kpachar** [comment]
@mapreri I'm facing this problem too. I've created another issue with the template filled up properly now. Hoping that it gets past the ansibot automated system. https://github.com/ansible/ansible/issues/23690

**Comment by mapreri** [comment]
@kpachar thanks, I subscribed it.
Although I leave this bug open (and I'm otherwise tempted to open a different bug about being able to specify extra options in ansible).

**Comment by ansibot** [comment]
cc @sashka
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your submission to Ansible. It means a lot to us that you've taken time to contribute.

Unfortunately, this issue has been open for some time while waiting for a contributor to take it up but there does not seem to have been anyone that did so. So we are going to close this issue to clear up the queues and make it easier for contributors to browse possible implementation targets.

However, we're absolutely always up for discussion. Because this project is very active, we're unlikely to see comments made on closed tickets and we lock them after some time. If you or anyone else has any further questions, please let us know by using any of the communication methods listed in the page below:

   * https://docs.ansible.com/ansible/latest/community/communication.html

In the future, sometimes starting a discussion on the development list prior to proposing or implementing a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible!

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)
